tests/test_integration.py

```diff
@@ -1338,3 +1340,41 @@ def test_issues_539_abi3_resolution():
     res.assert_success()
 
     subprocess.check_call([cryptography_pex, '-c', 'import cryptography'])
+
+
+@pytest.mark.skip("Acceptance test for landing https://github.com/pantsbuild/pex/issues/716.")
+def test_reproducible_build():
+
+  def assert_reproducible(args):
+    with temporary_dir() as td:
+      pex1 = os.path.join(td, '1.pex')
+      pex2 = os.path.join(td, '2.pex')
+      run_pex_command(args + ['-o', pex1])
+      run_pex_command(args + ['-o', pex2])
+      # First explode the PEXes to compare file-by-file for easier debugging.
+      with ZipFile(pex1) as zf1, ZipFile(pex2) as zf2:
+        zf1.extractall(path=os.path.join(td, "pex1"))
+        zf2.extractall(path=os.path.join(td, "pex2"))
+        for member1, member2 in zip(zf1.namelist(), zf2.namelist()):
+          assert filecmp.cmp(
+            os.path.join(td, "pex1", member1),
+            os.path.join(td, "pex2", member2),
+            shallow=False
+          )
+      # Then compare the original .pex files. This is the assertion we truly care about.
+      assert filecmp.cmp(pex1, pex2, shallow=False)
+
+  assert_reproducible([])
```

Comment:
You might want to isolate these asserts as separate tests for sanity sake.

Reply:
Good suggestion. Makes it easier to unskip tests, for example.